### ISSUE TITLE
CoreProtect Integration

### ISSUE THREAD

**Issue by script-head** [TotalFreedom/TotalFreedomMod#2066]
Currently, the official TF server is running CoreProtect. The TFM plugin on the server has some what of a bridge, but it's not very well made and has a lot of issues. I propose the idea of adding CoreProtect integration, but have CoreProtect a completely optional plugin to have as not all servers may want to use it.  Also, as the database does get rather large over time it needs to be cleared. Lastly, there should be a way to wipe only the flatlands. All of this is currently in the current TFM plugin on the server, but needs proper integration with better code.

**Comment by Wild1145** [comment]
I was sure we'd already had an issue for this, apparently not. 

The current Core Protect implementation method is not great, at all. I wont get into the discussion about it here, but the fact that this issue wasn't resolved prior to its addition to the server, is the reason I have continued to object its addition to the server. 

In terms of implementation, I'd simply have a config option so you can set if you plan to use the TFM-Core protect bridge (And I'd also suggest adding that for ALL the other plugins TFM works with like Essentials and WorldEdit). 

The database should only ever be selectively wiped based on the map that is being reset. Now we're on a VPS the initial plan was to build a custom panel to spec, that would allow this sort of functionality, however I dont know if MCMA supports this sort of functionality on a backup restore (I haven't used MCMA since we originally hosted CJF way back when and it was buggy as fuck to use, with no options for this sort of functionality)

**Comment by OxLemonxO** [comment]
@Wild1145 Can be done, but we'll need to write a custom plugin in C#.

**Comment by Wild1145** [comment]
So looking at this on a practical level. CoreProtect does not open source their code, or host any sort of Maven repo that we can use. As a result we cannot build CoreProtect into the TFM project, and certainly cannot currently built it through our CI tools. As such I am inclined to go with TFM not supporting Core Protect until such a time it is available to build automatically with, otherwise all that will happen is we waste all the hard work we have put into automating the build process. If however someone can find an official or even semi-official repo that is reliable that hosts CoreProtect, we can potentially look at supporting that feature. 

I think that sounds fair.

**Comment by lemonsked** [comment]
We can use something like this: https://github.com/rlf/XRayHunter/blob/master/coreprotect.xml
with the link:  https://addons-origin.cursecdn.com/files/2355/616/CoreProtect_2.14.2.jar
@Wild1145

**Comment by Wild1145** [comment]
@OxLemonxO I dont think that would work with the CI systems? As you're just building it basically as an after thought?

**Comment by lemonsked** [comment]
@Wild1145 Well, that project uses CI so, it should work in theory.
